forge/csrc/passes/commute_utils.cpp

```diff
@@ -898,16 +898,195 @@ bool can_commute_past_op(
     return (is_elementwise(op) and op->op_name() != "interleave") or is_quantization_ops(op);
 }
 
+void update_select_attr(
+    graphlib::OpNode *select_op,
+    int select_dim,
+    std::optional<int> begin,
+    std::optional<int> length,
+    std::optional<int> stride)
+{
+    log_trace("op name = {}", select_op->name());
+
+    if (select_op->op_name() != "select")
+    {
+        log_warning("update_select_attr called for a non-select operation: {}", select_op->op_name());
```

Comment:
Should this case be allowed? Do we need to assert instead of doing a silent exit? 

Same comment for other update function checks.

Reply:
It has been changed to assertion now.

Reply:
This pattern was used in the initial update_reshape_attr function [here](https://github.com/tenstorrent/tt-forge-fe/blob/e95d80e3e00a376540bd4261939b7063ce798a25/forge/csrc/passes/commute_utils.cpp#L903) , so it has been carried down in all newly created update functions.